### DISCUSSION TITLE
Modelling a concertina bellows

### DISCUSSION THREAD

**Discussion by rogerjames99** [Hopsan/hopsan#2221] [Q&A]
Firstly I am an absolute novice. I would like to model the action of an concertina instrument, to show the energy flow from the force applied to the bellows end caps, passing though the bellows reservoir, and pressing one or more key buttons. This is to show the force being applied to the bellows end caps and transferred  through  bellows enclosure, distributed via one or more reed valves, then applied to reed and converted to sound energy. I am not looking at detailed mechanical modelling. Just in the forces being applied at various points in the system.

My main objective is to clearly illustrate the distribution of energy to one or more reeds. i.e. That the more buttons that are pressed the less energy is given to an individual reed.

1. Is Hopsan a viable tool to do this?
2. Are there any other tools that provide an easier learning curve?
3. How do I get started?
4. Are then any more pneumatic example models other than the one in the supplied examples?

**Comment by robbr48** [comment]
Hi,
I think this could be possible, but as you might have noticed, the pneumatics library is still work in progress. We do not currently have any component for transforming mechanical forcee to pneumatic pressure, which I assume would be a requirement. We have plans to expand the library with both new components and more example models, but unfortunately I cannot give you a timeline for when such a component will be available. 

There might be some components available in the Modelica standard library, which is used in e.g. OpenModelica, but I would not say it has an easy learning curve.

I would say the approach would be to use a pneumatic piston component to simulate the key button and a variable volume for the bellow. It could also be possible to actually simulate the frequency of the tone by using the frequency analysis tools.
